### PR TITLE
[Merged by Bors] - Document options for !Sync types for Component and Resources

### PR DESCRIPTION
**Pull request by james7132**
# Objective
It's not clear to users how to handle `!Sync` types as components and resources in the absence of engine level support for them. 

## Solution
Added a section to `Component`'s and `Resource`'s type level docs on available options for making a type `Sync` when it holds `!Sync` fields, linking `bevy_utils::synccell::SyncCell` and the currently unstable `std::sync::Exclusive`.

Also added a compile_fail doctest that illustrates how to apply `SyncCell`. These will break when/if #6572 gets merged, at which point these docs should be updated.

### CODE REVIEW

**Review by hymm — APPROVED** [review]

**Review by devil-ira — APPROVED** [review]

File: crates/bevy_ecs/src/system/system_param.rs [hunk 1]
```diff
@@ -93,6 +93,33 @@ use std::{
...
+/// # use std::cell::RefCell;
+/// # use bevy_ecs::system::Resource;
+/// use bevy_utils::synccell::SyncCell;
+///
+/// // This will fail to compile since RefCell is !Sync.
+/// #[derive(Resource)]
+/// struct NotSync {
+///    counter: RefCell<usize>,
+/// }
+///
```
Comment (rodolphito) [inline crates/bevy_ecs/src/system/system_param.rs:113]: Could tighten the `compile_fail` test while ensuring the part that should compile compiles, and rendering the same in docs

```suggestion
/// ```
///
/// ```
/// # use std::cell::RefCell;
/// # use bevy_ecs::system::Resource;
/// # use bevy_utils::synccell::SyncCell;
```

File: crates/bevy_ecs/src/component.rs [hunk 2]
```diff
@@ -110,6 +110,33 @@ use std::{
...
+/// # use std::cell::RefCell;
+/// # use bevy_ecs::component::Component;
+/// use bevy_utils::synccell::SyncCell;
+///
+/// // This will fail to compile since RefCell is !Sync.
+/// #[derive(Component)]
+/// struct NotSync {
+///    counter: RefCell<usize>,
+/// }
+///
```
Comment (rodolphito) [inline crates/bevy_ecs/src/component.rs:130]:
```suggestion
/// ```
///
/// ```
/// # use std::cell::RefCell;
/// # use bevy_ecs::system::Resource;
/// # use bevy_utils::synccell::SyncCell;
```

**Review by rodolphito — APPROVED** [review]
Good to merge as is, comments are suggestions

**Review by TimJentzsch — APPROVED** [review]

**Review by JoJoJet — APPROVED** [review]

**Pull request closed without merging**